### PR TITLE
feat: add C `ndarray` API and refactor `blas/ext/base/dnansumkbn2`

### PR DESCRIPTION
**Pull request by headlessNode**
Resolves none.

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds C `ndarray` API and refactor/cleanup `blas/ext/base/dnansumkbn2`

## Related Issues

> Does this pull request have any related issues?

No. 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/lib/ndarray.js [hunk 1]
```diff
@@ -62,19 +62,20 @@ function dnansumkbn2( N, x, stride, offset ) {
 	var c;
 	var i;
 
+	ix = offsetX;
+	sum = 0.0;
+	ccs = 0.0; // second order correction term for lost low order bits
+	cs = 0.0; // first order correction term for lost low order bits
 	if ( N <= 0 ) {
 		return 0.0;
 	}
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/lib/ndarray.js]:
```suggestion
	sum = 0.0;
	ccs = 0.0; // second order correction term for lost low order bits
	cs = 0.0; // first order correction term for lost low order bits
	if ( N <= 0 ) {
		return 0.0;
	}
	ix = offsetX;
```

File: lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/lib/ndarray.js [hunk 2]
```diff
@@ -62,19 +62,20 @@ function dnansumkbn2( N, x, stride, offset ) {
 	var c;
 	var i;
 
+	sum = 0.0;
+	ccs = 0.0; // second order correction term for lost low order bits
+	cs = 0.0; // first order correction term for lost low order bits
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/lib/ndarray.js]:
```suggestion
	sum = 0.0;
```

File: lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/lib/ndarray.js [hunk 3]
```diff
@@ -62,19 +62,20 @@ function dnansumkbn2( N, x, stride, offset ) {
...
 		}
-		return x[ offset ];
+		sum = x[ ix ] * N;
+		return sum;
 	}
-	ix = offset;
-	sum = 0.0;
-	ccs = 0.0; // second order correction term for lost low order bits
-	cs = 0.0; // first order correction term for lost low order bits
 	for ( i = 0; i < N; i++ ) {
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/lib/ndarray.js:78]:
```suggestion
	ccs = 0.0; // second order correction term for lost low order bits
	cs = 0.0; // first order correction term for lost low order bits
	for ( i = 0; i < N; i++ ) {
```

File: lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/src/main.c [hunk 4]
```diff
@@ -32,59 +33,78 @@
...
+	CBLAS_INT i;
 	double cs;
 	double cc;
 	double v;
 	double t;
 	double c;
 
+	sum = 0.0;
+	ccs = 0.0; // second order correction term for lost lower order bits
+	cs = 0.0; // first order correction term for lost low order bits
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/src/main.c]:
```suggestion
	sum = 0.0;
```

File: lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/src/main.c [hunk 5]
```diff
@@ -32,59 +33,78 @@
...
-		ix = (1-N) * stride;
-	} else {
-		ix = 0;
+		sum = X[ ix ] * N;
+		return sum;
 	}
-	sum = 0.0;
-	ccs = 0.0; // second order correction term for lost lower order bits
-	cs = 0.0; // first order correction term for lost low order bits
 	for ( i = 0; i < N; i++ ) {
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/src/main.c:87]:
```suggestion
	ccs = 0.0; // second order correction term for lost lower order bits
	cs = 0.0; // first order correction term for lost low order bits
	for ( i = 0; i < N; i++ ) {
```

File: lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/lib/ndarray.js [hunk 6]
```diff
@@ -62,17 +62,17 @@ function dnansumkbn2( N, x, stride, offset ) {
 	var c;
 	var i;
 
+	sum = 0.0;
 	if ( N <= 0 ) {
 		return 0.0;
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/lib/ndarray.js]:
```suggestion
		return sum;
```

File: lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/lib/index.js [hunk 7]
```diff
@@ -38,7 +38,7 @@
 *
 * var x = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0, NaN, NaN ] );
 *
-* var v = dnansumkbn2.ndarray( 5, x, 2, 1 );
+* var v = dnansumkbn2.ndarray( x.length, x, 2, 1 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/lib/index.js]:
```suggestion
* var v = dnansumkbn2.ndarray( 5, x, 2, 1 );
```

**Review by Planeshifter — APPROVED** [review]
Thanks @headlessNode! Will merge shortly.

**Pull request merged**